R/teal_lockfile.R

```diff
@@ -39,50 +39,54 @@ teal_lockfile <- function() {
     }
   }
 
-  if (!(is_in_test() || is_r_cmd_check())) {
-    old_plan <- future::plan()
-    # If there is already a parallel (non-sequential) backend, reuse it.
-    if (inherits(old_plan, "sequential")) {
-      future::plan(future::multisession, workers = 2)
-    }
+  shiny::onStop(function() file.remove(lockfile_path))
 
-    lockfile_task <- ExtendedTask$new(create_renv_lockfile)
-    lockfile_task$invoke(close = inherits(old_plan, "sequential"), lockfile_path)
-    logger::log_trace("lockfile creation invoked.")
-  }
+  # Both capture.output are not needed if stdout = "|"
+  callr_output <-
+    utils::capture.output( # Needed to suppress: 'Opening fd 1' message
+      utils::capture.output( # Needed to suppress: 'PROCESS 'Rterm', running, pid' output
+        callr::r_bg(
+          func = create_renv_lockfile,
+          args = list(
+            lockfile_path = lockfile_path,
+            opts = options()
+          ),
+          # print results to the console
+          stdout = "",
+          # renv setup is orchestrated by its special S3 objects: renv::settings, renv::config and renv::paths
+          #     `package` parameter include `renv` namespace inside the environment of `func = create_renv_lockfile` fun
+          package = "renv"
+          # default env = NULL # means that callr process will use environmental variables
+          #     from the main session (parent process)
```

Comment:
why it is there? you provide the explanation to the commented out code?

Reply:
Looks alike, but not. It is an explanation of the defaults to understand how `env` is propagated to `r_bg`. It is not obvious from the documentation, we needed to dig to the docs of `processx::process`. Probably comment should be rephrased then to not suggest that it explains a commented-out code.

Reply:
if this is really needed: maybe uncomment this? - even if that's the default param value
right now it's look a little bit odd to me

Reply:
I think uncommenting the `env = NULL` and leaving the comment after it would be the cleanest

Reply:
For the `env` parameter I actually think we need to go with `env = Sys.getenv()` because `r_bg` by default uses `env = rcmd_safe_env()` 

```r
CYGWIN            R_TESTS          R_BROWSER        R_PDFVIEWER 
"nodosfilewarning"                 ""            "false"            "false"
```

`r_bg` also does not allow `env = NULL` even though `processx::process$new()` allows that. In `processx::process$new()` `env = NULL` means all parent system variables, but when `r_bg` limits them I think it's safer to use `Sys.getenv()`. 

`processx::process$new()` docs claims that you can append some variables to the list of all system variables
<img width="465" alt="image" src="https://github.com/insightsengineering/teal/assets/133694481/67f5b207-1d9f-4cc6-8fbe-43a5f63304b4">
but I don't think `callr` is appending anything with `current` value 
https://github.com/search?q=repo%3Ar-lib%2Fcallr%20%22current%22&type=code